realm/realm-library/src/androidTest/kotlin/io/realm/MixedBulkInsertsTests.kt

```diff
@@ -0,0 +1,652 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import io.realm.entities.*
+import io.realm.kotlin.createObject
+import io.realm.kotlin.where
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.junit.runner.RunWith
+import java.util.*
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+@RunWith(AndroidJUnit4::class)
+class MixedBulkInsertsTests {
+    private lateinit var realmConfiguration: RealmConfiguration
+    private lateinit var realm: Realm
+
+    @Rule
+    @JvmField
+    val folder = TemporaryFolder()
+
+    init {
+        Realm.init(InstrumentationRegistry.getInstrumentation().targetContext)
+    }
+
+    @Before
+    fun setUp() {
+        realmConfiguration = RealmConfiguration
+                .Builder(InstrumentationRegistry.getInstrumentation().targetContext)
+                .directory(folder.newFolder())
+                .schema(MixedNotIndexed::class.java,
+                        MixedIndexed::class.java,
+                        AllJavaTypes::class.java,
+                        MixedRealmListWithPK::class.java,
+                        MixedNotIndexedWithPK::class.java,
+                        PrimaryKeyAsString::class.java)
+                .build()
+
+        realm = Realm.getInstance(realmConfiguration)
+    }
+
+    @After
+    fun tearDown() {
+        realm.close()
+    }
+
+    private fun mixedTestValues() = RealmList(
```

Comment:
Nitpick: this could be a field instead.

Reply:
I like it to be a function because in some tests we mutate it by inserting managed/unmanaged realm models.